base_layer/core/src/chain_storage/postgres/migrations/2020-04-07-14-49_tx_outputs/up.sql

```diff
@@ -0,0 +1,15 @@
+create table tx_outputs(
+    hash TEXT NOT NULL PRIMARY KEY,
+    features_flags SMALLINT NOT NULL,
+    features_maturity BIGINT NOT NULL,
+    commitment TEXT NOT NULL,
+    proof BYTEA NULL,
+    spent BIGINT NOT NULL DEFAULT 0,
```

Comment:
The current idea behind this is, to make this the height at which this was spent. This will allow us to know when we can prune either the complete output, or just the rangeproof. 

I currently think we should rather replace this (and the inputs/output) tables and then add two fields, one for output block and one for input block.  This will force a 1:1 relationship to the an input and output of a "transaction output" as it can only be unspent once, and be spent once.

Reply:
Yeah, agree with this.

`created_in TEXT NOT NULL` to reference the block the output was created in. Definitely agree.
`spent_at_height BIGINT DEFAULT NULL` to reference the spending height.  Need to think about this, but so far I think this is correct. Using the hash _might_ be safer, but needs a bit more thought.

Reply:
hash should be safer, and should be unique...